<!DOCTYPE html> <html lang="en"> <head> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>Xinming Tu</title> <meta name="author" content="Xinming Tu"/> <meta name="description" content="A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design. "/> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"/> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"/> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="none" id="highlight_theme_light"/> <link rel="shortcut icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>⚛️</text></svg>"> <link rel="stylesheet" href="/assets/css/main.css"> <link rel="canonical" href="https://xinmingtu.github.io/"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark"/> <script src="/assets/js/theme.js"></script> <script src="/assets/js/dark_mode.js"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <div class="navbar-brand social"> <a href="mailto:%74%75%78%6D@%63%73.%77%61%73%68%69%6E%67%74%6F%6E.%65%64%75" title="email"><i class="fas fa-envelope"></i></a> <a href="https://scholar.google.com/citations?user=vE3n7lEAAAAJ" title="Google Scholar" target="_blank" rel="noopener noreferrer"><i class="ai ai-google-scholar"></i></a> <a href="https://github.com/XinmingTu" title="GitHub" target="_blank" rel="noopener noreferrer"><i class="fab fa-github"></i></a> <a href="https://twitter.com/TuXinming" title="Twitter" target="_blank" rel="noopener noreferrer"><i class="fab fa-twitter"></i></a> </div> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item active"> <a class="nav-link" href="/">About<span class="sr-only">(current)</span></a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">Blog</a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">Publications</a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">CV</a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fas fa-moon"></i> <i class="fas fa-sun"></i> </button> </li> </ul> </div> </div> </nav> </header> <div class="container mt-5"> <div class="post"> <header class="post-header"> <h1 class="post-title"> Xinming Tu </h1> <p class="desc"><a href="#">Univerisity of Washington</a>. Seattle.</p> </header> <article> <div class="profile float-right"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/prof_pic_PKU-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/prof_pic_PKU-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/prof_pic_PKU-1400.webp"></source> <img src="/assets/img/prof_pic_PKU.jpg" class="img-fluid z-depth-1 rounded" width="auto" height="auto" alt="prof_pic_PKU.jpg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> <div class="address"> <p>Weiming Lake, Bo Ya Ta, Peking University</p> </div> </div> <div class="clearfix"> <p>I am a second year PhD student at the <a href="https://www.cs.washington.edu" target="_blank" rel="noopener noreferrer">Paul Allen School of Computer Science and Engineering</a> at University of Washington supervised by <a href="http://saramostafavi.github.io" target="_blank" rel="noopener noreferrer">Sara Mostafavi</a>. Before my PhD, I earned BSs in Biology and Computer Science from Peking University. During my undergraduate, I did some bioinformatics research at <a href="http://www.gao-lab.org/index.php/en/home/" target="_blank" rel="noopener noreferrer">Gao Lab</a>. In addition, I also worked as a summer intern at <a href="http://zhaocenter.org" target="_blank" rel="noopener noreferrer">Zhao Lab</a> at Yale University and <a href="https://homes.cs.washington.edu/~swang/" target="_blank" rel="noopener noreferrer">Sheng Lab</a> at the University of Washington. I also spent six months on <a href="https://www.microsoft.com/en-us/research/group/machine-learning-research-group/" target="_blank" rel="noopener noreferrer">ML Group of Microsoft Research Asia</a> as a Full-time research Intern. Please see my <a href="https://xinmingtu.cn/assets/pdf/CV_homepage.pdf" target="_blank" rel="noopener noreferrer">CV</a> for more information.</p> <p>My research interest is using machine learning to decipher the principle of genomics. Recently my research focused on regulatory genomics under the single-cell context, including sequence to function, multi-omics integration, perturbation modeling.</p> </div> <div class="news"> <h2>News</h2> <div class="table-responsive"> <table class="table table-sm table-borderless"> <tr> <th scope="row">Feb 7, 2023</th> <td> I posted my third blog - <a href="https://xinmingtu.cn/blog/2023/LLM_science/" target="_blank" rel="noopener noreferrer">AI for General Science - Large language models for scientific hypothesis/research ideas generation</a> </td> </tr> <tr> <th scope="row">Jan 5, 2023</th> <td> This summer, I will join in the <a href="https://www.gene.com/scientists/our-scientists/aviv-regev" target="_blank" rel="noopener noreferrer">Aviv Regev</a>’s Group at Genentech as a Research Intern, work with <a href="https://romain-lopez.github.io" target="_blank" rel="noopener noreferrer">Romain Lopez</a>. </td> </tr> <tr> <th scope="row">Sep 14, 2022</th> <td> CLUE is accepted to NeurIPS 2022 as <strong>Oral presentation</strong>! See you in New Orleans! </td> </tr> <tr> <th scope="row">Jun 30, 2022</th> <td> I posted my second blog - <a href="https://xinmingtu.cn/blog/2022/Academic-resource/" target="_blank" rel="noopener noreferrer">Collection of some resources related to the PhD journey</a> </td> </tr> <tr> <th scope="row">Feb 16, 2022</th> <td> I gave a talk about our CLUE method on the 9th Winter q-bio conference! </td> </tr> <tr> <th scope="row">Feb 5, 2022</th> <td> I posted my first blog - <a href="https://xinmingtu.cn/blog/2022/CCA_dual_PCA/" target="_blank" rel="noopener noreferrer">Seurat CCA? It’s just a simple extension of PCA!</a> </td> </tr> </table> </div> </div> <div class="publications"> <h2>Selected publications</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge"><a href="https://nips.cc/" target="_blank" rel="noopener noreferrer">NeurIPS</a></abbr></div> <div id="10.1093/NeurIPS/bbab2dsf" class="col-sm-8"> <div class="title">Cross-Linked Unified Embedding for cross-modality representation learning</div> <div class="author"> <em>Xinming Tu*</em>, Zhi-Jie Cao*, Chen-Rui Xia, Sara Mostafavi†, and Ge Gao†</div> <div class="periodical"> <em>NeurIPS(Oral)</em> 2022 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="/assets/pdf/clue_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Multi-modal learning is essential for understanding information in the real world. Jointly learning from multi-modal data enables global integration of both shared and modality-specific information, but current strategies often fail when observa- tions from certain modalities are incomplete or missing for part of the subjects. To learn comprehensive representations based on such modality-incomplete data, we present a semi-supervised neural network model called CLUE (Cross-Linked Unified Embedding). Extending from multi-modal VAEs, CLUE introduces the use of cross-encoders to construct latent representations from modality-incomplete observations. Representation learning for modality-incomplete observations is common in genomics. For example, human cells are tightly regulated across multi- ple related but distinct modalities such as DNA, RNA, and protein, jointly defining a cell’s function. We benchmark CLUE on multi-modal data from single cell measurements, illustrating CLUE’s superior performance in all assessed categories of the NeurIPS 2021 Multimodal Single-cell Data Integration Competition. While we focus on analysis of single cell genomic datasets, we note that the proposed cross-linked embedding strategy could be readily applied to other cross-modality representation learning problems.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge"><a href="https://academic.oup.com/bioinformatics" target="_blank" rel="noopener noreferrer">Bioinformatics</a></abbr></div> <div id="10.1093/bioinformatics/btz768" class="col-sm-8"> <div class="title">Expectation pooling: an effective and interpretable pooling method for predicting DNA–protein binding</div> <div class="author"> Xiao Luo*, <em>Xinming Tu*</em>, Yang Ding, Ge Gao†, and Minghua Deng†</div> <div class="periodical"> <em>Bioinformatics</em> 2019 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1093/bioinformatics/btz768" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">HTML</a> <a href="/assets/pdf/epooling_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Convolutional neural networks (CNNs) have outperformed conventional methods in modeling the sequence specificity of DNA-protein binding. While previous studies have built a connection between CNNs and probabilistic models, simple models of CNNs cannot achieve sufficient accuracy on this problem. Recently, some methods of neural networks have increased performance using complex neural networks whose results cannot be directly interpreted. However, it is difficult to combine probabilistic models and CNNs effectively to improve DNA-protein binding predictions.In this article, we present a novel global pooling method: expectation pooling for predicting DNA-protein binding. Our pooling method stems naturally from the expectation maximization algorithm, and its benefits can be interpreted both statistically and via deep learning theory. Through experiments, we demonstrate that our pooling method improves the prediction performance DNA-protein binding. Our interpretable pooling method combines probabilistic ideas with global pooling by taking the expectations of inputs without increasing the number of parameters. We also analyze the hyperparameters in our method and propose optional structures to help fit different datasets. We explore how to effectively utilize these novel pooling methods and show that combining statistical methods with deep learning is highly beneficial, which is promising and meaningful for future studies in this field.All code is public in https://github.com/gao-lab/ePooling.Supplementary data are available at Bioinformatics online.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge"><a href="https://arxiv.org/" target="_blank" rel="noopener noreferrer">arXiv</a></abbr></div> <div id="tu2023data" class="col-sm-8"> <div class="title">What Should Data Science Education Do with Large Language Models?</div> <div class="author"> <em>Xinming Tu</em>, James Zou, Weijie J. Su†, and Linjun Zhang†</div> <div class="periodical"> <em>arXiv</em> 2023 </div> <div class="links"> <a href="https://arxiv.org/abs/2307.02792" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">HTML</a> </div> </div> </div> </li> </ol> </div> </article> </div> </div> <footer class="fixed-bottom"> <div class="container mt-0"> © Copyright 2023 Xinming Tu. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="noopener noreferrer">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" target="_blank" rel="noopener noreferrer">al-folio</a> theme. Last updated: July 22, 2023. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.6/dist/medium-zoom.min.js" integrity="sha256-EdPgYcPk/IIrw7FYeuJQexva49pVRZNmt3LculEr7zM=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js"></script> <script defer src="/assets/js/common.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script async src="https://www.googletagmanager.com/gtag/js?id=G-SQST9DCN49"></script> <script>function gtag(){window.dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-SQST9DCN49");</script> </body> </html>