---
layout: distill
title: A framework to understand regulatory genomics(To be finished)
description:
date: 2022-07-01

authors:
  - name: Xinming Tu
    url: "https://xinmingtu.cn"
    affiliations:
      name: University of Washington

bibliography: 2022-07-01-seq2func.bib

# Optionally, you can add a table of contents to your post.
# NOTES:
#   - make sure that TOC names match the actual section names
#     for hyperlinks within the post to work correctly.
#   - we may want to automate TOC generation in the future using
#     jekyll-toc plugin (https://github.com/toshimaru/jekyll-toc).
toc:
  - name: Background
  - name: The general framework
  - name: Two kinds of data
  - name: Summary
  - name: Case study
  - name: Related reviews
    # if a section has subsections, you can add them as follows:
    # subsections:
    #   - name: Example Child Subsection 1
    #   - name: Example Child Subsection 2
# Below is an example of injecting additional post-specific styles.
# If you use this post as a template, delete this _styles block.
---

## Background

<!-- 
Key question in genetics 
Some examples: GWAS, eQTL, CaQTL, regulatory genomics
From X, Y relationship or from cis-trans
Overview of this blog
 -->
The key question of genetics is studying the genetics process from genotype to phenotypes.

 genotype means the genome, the DNA sequences. Phenotype refers to an individual's observable traits, such as color, height, or weight.

However, We know there are some intermediate processes from genotype to phenotypes. That's the epigenotype. these include all epigenetics such as transcription factor binding, gene expression, histone modification, and Chromosome organization.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Geno2Pheno.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

Take previous research topics as examples, Genome-wide association studies (GWAS) study the association between Genotype and Phenotype. In GWAS, the genotype usually is the single-nucleotide polymorphisms (SNPs) and the phenotypes are traits like major human diseases.

The Expression quantitative trait loci(eQTL) study the association between Genotype and Epigenotype. In eQTL, the genotype is the variants on the genome and the Epigenotype is gene expression. Also in recent years, there is a new type of eQTL called Chromatin accessibility quantitative trait loci(caQTL). The only difference between eQTL and caQTL is the epigenotype is chromatin accessibility rather than gene expression.

What's regulatory genomics?
In my view, regulatory genomics is the study of the regulatory process of genes. It's all about the process from Genotype to epigenotype to phenotype. As for now, most of the regulatory genomics papers focus on the first part: Genotype to epigenotype.

| **Study**           | **Genotypes** | **Epigenotypes**        | **Phenotypes**  |
| ------------------- | ------------- | ----------------------- | --------------- |
| GWAS                | SNPs          | -                       | traits(disease) |
| eQTL                | Variants      | Gene expression         | -               |
| caQTL               | Variants      | Chromatin accessibility | -               |
| regulatory genomics | DNA sequences | Genomics tracks         | -               |

## The general framework

The general framework of sequence to function/activity is as follows:

The model takes the DNA sequence as input and outputs the activity of the different genomics tracks, such as gene expression, chromatin accessibility, 3D genome structure, etc.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Framework.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

The formulation in math is very clear: $$Y = f(S_i)$$

## Two kinds of data

<!-- 
Genome sequences
Random sequences
-->

So the next key question is what data should we use to train the model?
There are two kinds of data: the first one is the DNA sequence from the genome, and use sequencing technology to get the nature readout from cells. Another one is using synthetic biology to generate the random sequences and then using experimental technology to get the measurement from a fixed system.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/two_data.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

There are some advantages and disadvantages of the two kinds of data. We will talk about that more in our next section.

### The genome sequences

<!--
Only sequence as input
Other modalities as input
    - for cell type-specific prediction
    - for multi-layer relationships? 
Pros and cons 
 -->

There are more than six billion base-pair of DNA in the human genome, which can be easily used as input. And we can use experimental technology to get different nature measurements from the cells, such as gene expression, chromatin accessibility, 3D genome structure, etc.

We can take the DNA sequence as input and use the model to predict the activity of the different genomics tracks.

- Predict **Gene expression** from sequences: Basenji<d-cite key="kelley2018sequential"></d-cite>, ExPecto<d-cite key="zhou2018deep"></d-cite>, Enformer<d-cite key="avsec2021effective"></d-cite>
- Predict **chromatin accessibility** from sequences: Basset<d-cite key="kelley2016basset"></d-cite>, DanQ<d-cite key="quang2016danq"></d-cite>
  , AI-TAC<d-cite key="maslova2020deep"></d-cite>, AMBER<d-cite key="zhang2021automated"></d-cite>
- Predict the **3D structure of the genome** from sequences:Akita<d-cite key="fudenberg2020predicting"></d-cite>, DeepC<d-cite key="schwessinger2020deepc"></d-cite>, C.Origami<d-cite key="tan2022cell"></d-cite>, Orca<d-cite key="zhou2022sequence"></d-cite>
- Predict the **protein binding** from sequences: DeepBind<d-cite key="alipanahi2015predicting"></d-cite>, DeepSEA<d-cite key="zhou2015predicting"></d-cite>, BPnet<d-cite key="avsec2021base"></d-cite>

Also beyond predicting genomics tracks, we can take other genomics tracks as input too.

- Predict the **gene expression** from sequences and **3D genome structure**: GraphReg<d-cite key="karbalayghareh2022chromatin"></d-cite>
- Predict the **Protein binding** from DNA sequences and **Chromatin accessibility**: Leopard<d-cite key="li2021fast"></d-cite>, maxATAC<d-cite key="cazares2022maxatac"></d-cite>

In this way, we can extend the method to predict cell-type-specific genomics tracks. The reason is that we believe that the cell-type-specific genomics tracks can be represented by the cell-type-specific DNA genomics tracks so if we take that as input, we can predict. Also study the multi-layer relationships between different genomics tracks.


For example, based on our knowledge of protein-DNA binding. There are two requirements: The DNA Sequence should contain Motifs and this region should be accessible by the protein. So It needs two kinds of information.

Figure summary:

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Genome_examples.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

Using genome sequences, we can use a **long sequence** to predict the activity of the different genomics tracks. Even across the whole chromosome<d-cite key="zhou2022sequence"></d-cite>. Therefore, we can study the complex mechanism of the genome including almost all possible regulation processes. (Promoter, enhancer, repressor, long-range loop, compartments change). Also, we can take the data from different cell types as input, so it's possible to predict the cell-type-specific genomics tracks.

Meanwhile, there are also some disadvantages to using genome sequences.

- The most important one is that the sample size is limited. For example, if we want to predict gene expression, the sample size won't be larger than the number of genes$$3000 – 40000$$. If we want to predict chromatin accessibility, the sample size will be larger than the number of open chromatin regions. If we don't consider the overlapping between different inputs, the sample size $$N$$ should be smaller than $$\frac{L_{genome}}{L_{input}}$$. It's the natural limitation of the sample size.
- Another limitation is that the input genome is from the genome which means it's a steady state in evolution. We cannot get any flexibility in the input genome through the population level there will be some variants.
- Pros:
  - Long sequences(L)
  - Complex mechanism
  - Enhancer, Promoter, 3D structure
  - Different Cell types
- Cons: - Small sample size $$ n $$
  - Genes ~ , $$ N \times L < total length $$
    Steady-state
  - Limited space, It’s the evolution of steady status. Correlated to each other

### The synthesis sequences

If we synthesize the random sequences and introduce these sequences into a different position in the genome, we can study the function of this element in a very flexible way. There are also some papers working on this topic.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/randome_sequence.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

We can use random sequences as **Promoter** to study the mechanism of the promoter. We can also use random sequences as 3'UTR or 5'UTR<d-cite key="sample2019human, cuperus2017deep"></d-cite> to study the mechanism of the 5'UTR or 3'UTR. Furthermore, we can use random sequences as enhancers and **Promoter\*\***<d-cite key="sahu2022sequence"></d-cite>

There are some advantages and disadvantages of random sequences. As for the advantages, we can generate a large number of random sequences to get a large sample size( even more than 300 million). Also, there is no constraint on the sequences, the sequence space is flexible. we can synthesize any sequences we want and do experiments to validate the model predictions.

However, We only can study **the simple system**(such as yeast) in this way, and the sequence will be **short**(usually the length is not longer than 200). As a result, the mechanism will also be very \*\*local and simple.

- Pros:
  - Large sample size
  - Flexible sequences
  - Experimental validation
- Cons:
  - Simple system
  - Short sequences
  - Local mechanism

## From model to new biology

Now, we have the data and the model. What's next? The final goal is to find the new biology knowledge or principle. So we need to dig into the model and find the new biology.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/new_biology.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

### Decipher the regulatory process from the model

The first kind of application is to decipher the regulatory process from the model.

After the training, the model already knows the regulatory process. So we can use the explainable AI method(Such as DeepLift<d-cite key="shrikumar2017learning"></d-cite>, SHAPLY<d-cite key="lundberg2017unified"></d-cite>) to decipher the regulatory process.

We can learn the regulatory elements from the model(TF-Modisco<d-cite key="shrikumar2018technical"></d-cite>) and also get the relationship between the regulatory elements.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/decipher_model.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

### Simulator to optimize and do experiments in virtual

The second kind of application is using the model to simulate the regulatory process virtually and we can also use the simulator to optimize the sequence and do engineering experiments.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/model_simulator.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

We have a trusty simulator(Simulator) and we can use the simulator to do experiments virtually and introduce mutations/perturbation to study the effect of variants which can even extend to large structure variants (such as deletion, insertion, and duplication). Also, we can use the simulator to do some evolutionary studies. We can mimic the evolution of the regulatory process.

## Summary

So in this post, we have learned the following:
There are two kinds of data: The genome sequences, The synthesis sequences and two kinds of ways to get new biology: Deciphering the regulatory process from the model, Simulator to optimize and do experiments in virtual

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Summary.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

## Case study

In this section, we list some papers as case studies to better understand the framework.

## Related reviews

There are also some really great related reviews:
If you want to know more about regulatory genomics, you can read the following reviews:

- Decoding disease: from genomes to networks to phenotypes<d-cite key="wong2021decoding"></d-cite>
