---
layout: distill
title: A framework to understand regulatory genomics(To be finished)
description:
date: 2022-07-01

authors:
  - name: Xinming Tu
    url: "https://xinmingtu.cn"
    affiliations:
      name: University of Washington

bibliography: 2022-07-01-seq2func.bib

# Optionally, you can add a table of contents to your post.
# NOTES:
#   - make sure that TOC names match the actual section names
#     for hyperlinks within the post to work correctly.
#   - we may want to automate TOC generation in the future using
#     jekyll-toc plugin (https://github.com/toshimaru/jekyll-toc).
toc:
  - name: Background
  - name: The general framework
  - name: Two kinds of data
  - name: Summary
  - name: Case study
  - name: Related reviews
    # if a section has subsections, you can add them as follows:
    # subsections:
    #   - name: Example Child Subsection 1
    #   - name: Example Child Subsection 2
# Below is an example of injecting additional post-specific styles.
# If you use this post as a template, delete this _styles block.
---

## Background

The key question of genetics is

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Geno2Pheno.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

previous works:
Genotype to phenotype

GWAS, e-QTL, ca-QTL

What's the regulatory genomics?

## The general framework

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Framework.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

## Two kinds of data

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/two_data.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

### The genome sequences

Bassenji <d-cite key="kelley2018sequential"></d-cite>

ExPecto <d-cite key="zhou2018deep"></d-cite>

Enformer <d-cite key="avsec2021effective"></d-cite>
GraphReg <d-cite key="karbalayghareh2022chromatin"></d-cite>
Akita <d-cite key="fudenberg2020predicting"></d-cite>
DeepC <d-cite key="schwessinger2020deepc"></d-cite>
C.Origami <d-cite key="schwessinger2020deepc"></d-cite>
Orca <d-cite key="zhou2022sequence"></d-cite>
Basset <d-cite key="kelley2016basset"></d-cite>
DanQ <d-cite key="quang2016danq"></d-cite>
AI-TAC <d-cite key="maslova2020deep"></d-cite>
AMBER <d-cite key="zhang2021automated"></d-cite>
Leopard <d-cite key="li2021fast"></d-cite>
maxATAC <d-cite key="cazares2022maxatac"></d-cite>
DeepBind <d-cite key="alipanahi2015predicting"></d-cite>
DeepSEA <d-cite key="zhou2015predicting"></d-cite>
BPnet <d-cite key="avsec2021base"></d-cite>

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Genome_examples.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

- Pros:
  - Long sequences(L)
  - Complex mechanism
  - Enhancer, Promoter, 3D structure
  - Different Cell types
- Cons: - Small sample size $$ n $$
  - Genes ~ $$ 3000 – 40000 $$, $$ N \times L < total length $$
  - Steady state
  - Limited space, It’s the evolution steady status. Correlated to each other

### The random sequences

Previous works

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/randome_sequence.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

Promoter <d-cite key="de2020deciphering, vaishnav2022evolution"></d-cite>
3'UTR <d-cite key="griesemer2021genome"></d-cite>
5'UTR <d-cite key="sample2019human, cuperus2017deep"></d-cite>
Enhancer and Promoter <d-cite key="sahu2022sequence"></d-cite>

- Pros:
  - Large sample size
  - Flexible sequences
  - Experimental validation
- Cons:
  - Simple system
  - Short sequences
  - Local mechanism

## From model to new biology

Now, we have the data and the model. What's next? The final goal is to find the new biology knowledge or principle. So we need to dig into the model and find the new biology.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/new_biology.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

### Decipher the regulatory process from the model

The first kind of application is to decipher the regulatory process from the model.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/decipher_model.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

### Simulator to optimize and do experiments in virtual

The second kind of application is using the model to simulate the regulatory process in virtual and we can also use the simulator to optimize the sequence and do engineering experiments.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/model_simulator.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

## Summary

So in this post, we have learned the following:
There are two kinds of data: The genome sequences, The random sequences and two kinds of applications: Decipher the regulatory process from the model, Simulator to optimize and do experiments in virtual

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Summary.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

## Case study

In this section, we list some papers as the case study to better understand the framework.

## Related reviews

There are also some really great related reviews:
If you want to know more about the regulatory genomics, you can read the following reviews:
