---
layout: distill
title: A framework to understand regulatory genomics(To be finished)
description:
date: 2022-07-01

authors:
  - name: Xinming Tu
    url: "https://xinmingtu.cn"
    affiliations:
      name: University of Washington

bibliography: 2022-07-01-seq2func.bib

# Optionally, you can add a table of contents to your post.
# NOTES:
#   - make sure that TOC names match the actual section names
#     for hyperlinks within the post to work correctly.
#   - we may want to automate TOC generation in the future using
#     jekyll-toc plugin (https://github.com/toshimaru/jekyll-toc).
toc:
  - name: Background
  - name: The general framework
  - name: Two kinds of data
  - name: Summary
  - name: Case study
  - name: Related reviews
    # if a section has subsections, you can add them as follows:
    # subsections:
    #   - name: Example Child Subsection 1
    #   - name: Example Child Subsection 2
# Below is an example of injecting additional post-specific styles.
# If you use this post as a template, delete this _styles block.
---

## Background

<!-- 
Key question in genetics 
Some examples: GWAS, eQTL, CaQTL, regulatory genomics
From X, Y relationship or from cis-trans
Overview of this blog
 -->

One of the central questions in genetics is understanding the relationship between genotype and phenotype. Genotype refers to an individual's genome, or their complete set of DNA sequences, while phenotype refers to observable traits such as physical characteristics and disease states. A deeper understanding of the relationship between genotype and phenotype requires an examination of the intermediate processes that link the two. One of the most important of these processes is gene regulation, which is a multi-layered, context-dependent process. It involves interactions between various elements, including regulatory elements, the abundance of transcription factors, chromatin states, and various epigenetic modifications. Together, these intermediate processes and variables can be referred to as epigenotype/epigenomics, which plays a crucial role in determining an individual's observable traits



<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Geno2Pheno.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

Take previous research topics as examples, Genome-wide association studies (GWAS) are a powerful tool for understanding the relationship between genotype and phenotype. These studies typically focus on the association between single-nucleotide polymorphisms (SNPs) and various human disease states. Another area of research that examines this relationship is expression quantitative trait loci (eQTL) studies, which investigate the association between genetic variants and epigenotype, specifically gene expression. In recent years, a new type of eQTL called chromatin accessibility quantitative trait loci (caQTL) has emerged, which examines the relationship between genetic variants and chromatin accessibility. These statistical genetics approaches use regression models based on predefined features, such as different alleles, to establish associations. However, this approach has a limitation that it can only introduce mutations that are present in the training data, and the input features of the model are determined by the training data.

**What's regulatory genomics?** In my view, Regulatory genomics is a field of study that aims to understand the underlying mechanisms of gene regulation at a higher resolution. This is achieved by using computational and experimental approaches to predict the epigenotypes from DNA sequences, which allows for a more flexible exploration of the genotype-phenotype relationship than traditional statistical genetics methods. Regulatory genomics often uses bioinformatics tools and techniques such as genome-wide ChIP-seq, RNA-seq, and ATAC-seq, to identify and analyze different types of regulatory elements, including transcription factors, enhancers, and promoter regions, with the goal of understanding how these elements interact to regulate gene expression and ultimately affect the phenotype.

| **Study**           | **Genotypes** | **Epigenotypes**        | **Phenotypes**  |
| ------------------- | ------------- | ----------------------- | --------------- |
| GWAS                | SNPs          | -                       | traits(disease) |
| eQTL                | Variants      | Gene expression         | -               |
| caQTL               | Variants      | Chromatin accessibility | -               |
| Regulatory genomics | DNA sequences | Genomics tracks         | -               |

In this blog, I will provide an in-depth overview of the field of regulatory genomics. I will start by introducing a general framework that outlines the key concepts and principles of the field. Afterward, I will delve into specific examples by discussing two types of data that are commonly used in regulatory genomics: genome sequences and synthetic sequences. Additionally, I will highlight some of the recent research papers that have contributed to the progress and advancements in the field. To conclude, I will explore the different ways in which the models developed in regulatory genomics can be applied to gain new insights into the mechanisms of gene regulation and the relationship between genotype and phenotype.

## The general framework

The general framework of regulatory genomics is the prediction of functional and activity outcomes from DNA sequences. This is achieved by using computational models that take the DNA sequence as input $$X$$ and predict the activity of various genomics tracks $$Y$$, such as gene expression, chromatin accessibility, and 3D genome structure.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Framework.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

## Two kinds of data

<!-- 
Genome sequences
Random sequences
-->

A crucial aspect of developing models in regulatory genomics is selecting the appropriate data source. Two commonly used data sources in this field are genome sequences and synthetic sequences. Genome sequences are obtained through sequencing technology and provide a snapshot of the genomic landscape in a natural context. They are useful for studying the effects of genetic variations on gene regulation. Synthetic sequences, on the other hand, are generated using synthetic biology techniques and are used to study the effects of genetic variations in a specific region in a controlled environment, typically in a specific cell line. These data sources provide a unique opportunity to study the effects of genetic modifications on gene regulation.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Two_data.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

Both types of data sources have their own advantages and limitations, which will be discussed in further detail in the later sections of the blog post

### The genome sequences

<!--
Only sequence as input
Other modalities as input
    - for cell type-specific prediction
    - for multi-layer relationships? 
Pros and cons 
 -->
The human genome contains more than six billion base-pairs, which makes it an easily accessible data source for inputting into models. Additionally, various experimental technologies can be utilized to obtain different types of measurements from cells in various states and functions.

Models in regulatory genomics take DNA sequences as input and predict the activity of different genomics tracks, such as gene expression, chromatin accessibility, 3D genome structure, protein binding, and more. 

#### Sequence as input

There are various works that focus on predicting specific genomic tracks, such as:

- Predict **protein binding** from sequences: DeepBind<d-cite key="alipanahi2015predicting"></d-cite>, DeepSEA<d-cite key="zhou2015predicting"></d-cite>, BPnet<d-cite key="avsec2021base"></d-cite>
- Predict **gene expression** from sequences: Basenji<d-cite key="kelley2018sequential"></d-cite>, ExPecto<d-cite key="zhou2018deep"></d-cite>, Enformer<d-cite key="avsec2021effective"></d-cite>
- Predict **chromatin accessibility** from sequences: Basset<d-cite key="kelley2016basset"></d-cite>, DanQ<d-cite key="quang2016danq"></d-cite>
  , AI-TAC<d-cite key="maslova2020deep"></d-cite>, AMBER<d-cite key="zhang2021automated"></d-cite>
- Predict **3D structure of the genome** from sequences:Akita<d-cite key="fudenberg2020predicting"></d-cite>, DeepC<d-cite key="schwessinger2020deepc"></d-cite>, Orca<d-cite key="zhou2022sequence"></d-cite>

#### Sequence and other modalities as input

In addition to using DNA sequences as the sole input, other genomics tracks can also be incorporated as inputs. For example:

- Predict **protein binding** from DNA sequences and **Chromatin accessibility**: Leopard<d-cite key="li2021fast"></d-cite>, maxATAC<d-cite key="cazares2022maxatac"></d-cite>
- Predict **gene expression** from sequences and **3D genome structure**: GraphReg<d-cite key="karbalayghareh2022chromatin"></d-cite>
- Predict **3D genome structure** from sequences and **Chromatin accessibility/Protein binding**: C.Origami<d-cite key="tan2022cell"></d-cite>

The main motivations behind incorporating other modalities as inputs are: (1)The ability to predict cell-type specific activity directly. (2)The ability to study multi-layer relationships between different genomics tracks.

<!-- The previous methods only use the DNA sequences as the input, which means the input of different cells will be the same(the same genome), so they have to use a multi-task setting to predict cell-type-specific activity. However, if we incorporate other modalities as the input, we can predict the cell-type-specific activity directly<d-cite key="li2021fast"></d-cite>. For example, based on our knowledge of protein-DNA binding, There are two requirements: The DNA Sequence should contain Motifs and this region should be accessible by the protein. So if the model has these two kinds of information, it can be generalized to new cell types. It's possible to study the multi-layer relationships between different genomics tracks in this setting<d-cite key="karbalayghareh2022chromatin"></d-cite>. -->

Previous methods that rely solely on DNA sequences as input can only provide a general understanding of gene regulation, as the input for different cells is the same (the same genome). To account for cell-type specific activity, these methods often use a multi-task setting. On the other hand, incorporating other modalities as input, such as chromatin accessibility or protein-DNA binding, allows for direct prediction of cell-type specific activity. For example, if we know that a specific DNA sequence contains a motif and is accessible to a certain protein, we can generalize this understanding to other cell types<d-cite key="li2021fast"></d-cite>. Additionally, incorporating other modalities as input allows for the study of multi-layer relationships between different genomics tracks, providing a more comprehensive understanding of gene regulation<d-cite key="karbalayghareh2022chromatin"></d-cite>.

Figure summary:

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Genome_examples.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

#### Pros and cons

Using genome sequences as a data source in regulatory genomics allows for the prediction of the activity of different genomics tracks across long sequences, even entire chromosomes<d-cite key="zhou2022sequence"></d-cite>. This allows for the study of complex mechanisms of the genome, including various types of regulatory processes such as promoters, enhancers, repressors, long-range loops, and changes in compartments. Additionally, by using data from different cell types, it is possible to predict cell-type-specific genomics tracks.

However, there are also some disadvantages to using genome sequences as a data source. One of the main limitations is the limited sample size. For example, when predicting gene expression, the sample size is limited to the number of genes. Similarly, when predicting chromatin accessibility, the sample size is limited to the number of open chromatin regions. Another limitation is that the genome is in a state of constant evolution, which limits the diversity of sequences.

- Pros of using genome sequences as a data source in regulatory genomics:
  - Allows for the prediction of activity across **long sequences** even entire chromosomes
  - Enables the study of **complex mechanisms** of the genome, including various types of regulatory processes such as enhancers, promoters, and 3D structure
  - Provides the ability to predict **cell-type-specific** genomics tracks
- Cons of using genome sequences as a data source in regulatory genomics:
  - **Limited sample size**: The number of samples is restricted by the number of genes or open chromatin regions.
  - The genome is in a state of constant evolution, which **limits the diversity of sequences**

### The synthesis sequences

If we synthesize the sequences and introduce these sequences into a different position in the genome, we can study the function of this element in a very flexible way. There are also some papers working on this topic.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/randome_sequence.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

We can use synthesize sequences as **Promoter** to study the mechanism of the promoter. We can also use random sequences as 3'UTR or 5'UTR<d-cite key="sample2019human, cuperus2017deep"></d-cite> to study the mechanism of the 5'UTR or 3'UTR. Furthermore, we can use random sequences as enhancers and **Promoter**<d-cite key="sahu2022sequence"></d-cite>

There are some advantages and disadvantages of random sequences. As for the advantages, we can generate a large number of random sequences to get a large sample size(even more than 300 million). Also, there is no constraint on the sequences, the sequence space is flexible. we can synthesize any sequences we want and do experiments to validate the model predictions.

However, We only can study **the simple system**(such as yeast) in this way, and the sequence will be **short**(usually the length is not longer than 300). As a result, the mechanism will also be very **local** and **simple**.

- Pros:
  - Large sample size
  - Flexible sequences
  - Experimental validation
- Cons:
  - Simple system
  - Short sequences
  - Local mechanism

## From model to new biology

Now, we have the data and the model. The final goal is to find new biology insights. So we need to dig into the model and find the new biology. There are two ways to do this: **Deciphering** and **Engineering**.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/new_biology.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

### Decipher the regulatory process from the model

The first kind of application is to decipher the regulatory process from the model.

After the training, the model already knows the regulatory process. So we can use the explainable AI method(Such as DeepLift<d-cite key="shrikumar2017learning"></d-cite>, SHAPLY<d-cite key="lundberg2017unified"></d-cite>) to decipher the regulatory process<d-cite key="novakovsky2022obtaining"></d-cite>.

We can learn the regulatory elements from the model(TF-Modisco<d-cite key="shrikumar2018technical"></d-cite>) and also get the relationship between the regulatory elements.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Decipher_model.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

### Simulator to optimize and do experiments in virtual

The second kind of application is using the model to simulate the process virtually and we can also use the simulator to optimize the sequence and do engineering experiments.

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/model_simulator.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

We have a trusty simulator(Simulator) and we can use the simulator to do experiments virtually and introduce mutations/perturbation to study the effect of variants which can even extend to large structure variants (such as deletion, insertion, and duplication). Also, we can use the simulator to do some evolutionary studies. We can mimic the evolution of the regulatory process.

## Summary

So in this post, we have learned the following:
There are two kinds of data: The genome sequences/the synthesis sequences and two kinds of ways to get new biology: Deciphering the regulatory process from the model, Simulator to optimize and do experiments in virtual

<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/seq2func/Summary.pdf" class="img-fluid rounded z-depth-1" %}
    </div>
</div>

## Case study

In this section, we list some papers as case studies to better understand the framework.

## Related reviews

There are also some really great related reviews:
If you want to know more about regulatory genomics, you can read the following reviews:

- Decoding disease: from genomes to networks to phenotypes<d-cite key="wong2021decoding"></d-cite>
- Obtaining genetics insights from deep learning via explainable artificial intelligence<d-cite key="novakovsky2022obtaining"></d-cite>